src/storage/store/node_group.cpp

```diff
@@ -193,11 +196,107 @@ NodeGroupScanResult NodeGroup::scan(Transaction* transaction, TableScanState& st
         chunkedGroupToScan.scan(transaction, state, nodeGroupScanState, rowIdxInChunkToScan,
             numRowsToScan);
     }
-    const auto startRow = nodeGroupScanState.numScannedRows;
-    nodeGroupScanState.numScannedRows += numRowsToScan;
+    const auto startRow = nodeGroupScanState.nextRowToScan;
+    nodeGroupScanState.nextRowToScan += numRowsToScan;
     return NodeGroupScanResult{startRow, numRowsToScan};
 }
 
+NodeGroupScanResult NodeGroup::scan(Transaction* transaction, TableScanState& state,
```

Comment:
Yes, I can try and add to the new tests. I'd been trying to merge this new function with the existing one so that it would be better covered, but I had run into some problems with that.

Reply:
Have you added tests for these? I still see many of these lines not tested.

Reply:
I didn't actually add any new tests. The bulk of what wasn't covered, was [this block](https://app.codecov.io/gh/kuzudb/kuzu/pull/4453#e5237d6f81b6af3f8b03ca119c6a2d70-R208), which was duplicated from elsewhere in the file. I've since merged them into a single function [here](https://github.com/kuzudb/kuzu/pull/4453/files#diff-776a83f6364f9ed72b1143204edde01675e100adc037f35e605dd18c038321a4R141-R169), but code-cov doesn't appear to actually update the coverage report when the PR is updated, so it still is showing those original lines as uncovered.